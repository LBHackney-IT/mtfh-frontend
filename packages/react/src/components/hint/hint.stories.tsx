import React from "react";

import { Hint } from "./hint";

export default {
  title: "Hint",
  component: Hint,
};

export const Base = () => {
  return <Hint>A descriptive hint</Hint>;
};

export const PolymorphicHint = () => {
  return <Hint as="p">A descriptive hint</Hint>;
};
