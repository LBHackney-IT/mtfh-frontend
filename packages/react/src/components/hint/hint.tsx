import React, { forwardRef } from "react";
import type * as Polymorphic from "@radix-ui/react-polymorphic";
import cn from "classnames";

import styles from "./styles.module.scss";

export type HintComponent = Polymorphic.ForwardRefComponent<"span">;

export const Hint: HintComponent = forwardRef(function Hint(
  { as: HintComp = "span", className, ...props },
  ref
) {
  return (
    <HintComp
      ref={ref}
      className={cn(styles.govukHint, styles.lbhHint, className)}
      {...props}
    />
  );
});
