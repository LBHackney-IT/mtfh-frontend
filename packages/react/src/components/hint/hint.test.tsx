import React from "react";
import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Hint } from "./hint";

test("it renders the hint", async () => {
  const { container } = render(<Hint>Hint</Hint>);
  expect(container).toMatchSnapshot();
  await testA11y(container);
});
