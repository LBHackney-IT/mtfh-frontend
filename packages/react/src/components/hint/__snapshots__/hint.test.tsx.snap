// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the hint 1`] = `
<div>
  <span
    class="govukHint lbhHint"
  >
    Hint
  </span>
</div>
`;
